src/Symfony/Component/ExpressionLanguage/Tests/ExpressionLanguageTest.php

```diff
@@ -237,6 +237,37 @@ public function testRegisterAfterEval($registerCallback)
         $registerCallback($el);
     }
 
+    public function testBadProperty()
+    {
+        $this->expectException(\RuntimeException::class);
+        $this->expectExceptionMessageMatches('/Unable to get property "\w+" of object "\w+"./');
+        $el = new ExpressionLanguage();
+        $el->evaluate('foo.bar', ['foo' => new \stdClass()]);
+    }
+
+    public function testBadPropertyNullSafe()
+    {
+        $this->expectException(\RuntimeException::class);
+        $this->expectExceptionMessageMatches('/Unable to get property "\w+" of object "\w+"./');
+        $el = new ExpressionLanguage();
+        $el->evaluate('foo?.bar', ['foo' => new \stdClass()]);
+    }
+
+    public function testBadObjectGetProperty()
+    {
+        $this->expectException(\RuntimeException::class);
+        $this->expectExceptionMessageMatches('/Unable to get property "\w+" of non-object "\w+"./');
+        $el = new ExpressionLanguage();
+        $el->evaluate('foo.bar', ['foo' => null]);
+    }
+
+    public function testBadObjectGetPropertyNullSafe()
+    {
+        $el = new ExpressionLanguage();
+        $result = $el->evaluate('foo?.bar', ['foo' => null]);
```

Comment:
please add a test case were the foo variable is not declared,
and also one where the nullsafe is nested foo.bar?.baz, where only foo exists and is not null